Add fieldName to the returned file object in processRequest

lorenzodejong opened the pull request:
When processing a dynamic list of files with corresponding ID's in a GraphQL resolver the filename is not sufficient for determining the reference. This PR adds the `fieldName` property to the returned file object which can be used to assign arbitrary identifiers from the GraphQL query/mutation. This comes in handy when you have a dynamic schema of expected files which can't be expressed using a static GraphQL query. An example (my personal use case) is a user-generated registration form with an arbitrary amount of upload fields.

Example query (multipart form-data):
- operations: `{"query":"mutation UserUploadImage($files: [Upload!]!) { UploadImage(files: $files) }", "variables": { "files": [] }}`
- map: `{"some-id": ["variables.files.0"], "another-id": ["variables.files.1"]}`
- some-id: `File`
- another-id: `File`

Example GraphQL resolver:
```javascript
async function uploadFile(file) {
	return new Promise(async (resolve, reject) => {
		const { createReadStream, fieldName } = await file;

		const id = fieldName;

		const stream = createReadStream();
		const out = require('fs').createWriteStream(id);

		stream.pipe(out);

		out.on('finish', () => {
			resolve({ id });
		});

		out.on('error', (error) => {
			reject(error);
		});
	});
}

export async function resolve(_parent, { files }, _ctx) {
	// Upload all files, return an array of objects including identifiers (fieldName)
	const uploads = await Promise.all(files.map(uploadFile));

	// Example DB call which stores references to the files using the identifier (fieldName)
	await db.user.update({
		data: {
			uploads
		}
	});

	return true;
}
```

koresar commented:
PR looks good. We need unit tests updated too. Could you please fix them up in this file too?
https://github.com/flash-oss/graphql-upload-minimal/blob/f32a6b508853c27f54fe07ca3fd5ca2c738d005a/test/public/processRequest.test.js#L40

koresar commented:
Also, README would need to be updated too if possible.
And JSDoc: https://github.com/flash-oss/graphql-upload-minimal/blob/master/public/index.js
And README re-generated with the latest JSDoc changes. :)

In other words, I'd recommend looking up the repo by the word "mimetype" and insert the new field right next to it pretty much everywhere (except auto-generated section of the README).

lorenzodejong commented:
I've modified the PR to fix the unit tests and documentation. Additionally i've also added an example in the docs on how to process multiple file uploads in the GraphQL resolver using the new `fieldName` property.

koresar commented:
Sorry for the delay. Merged your PR to a new branch. I'll review in on my computer and pushing ASAP.

Thank you very much for this nice feature.
I've just granted you write access to this repo. Feel free to push commits directly to it. :)

koresar commented:
published as v1.3

lorenzodejong commented:
Awesome! Thanks for the review :).
